SampleGallery/Samples/SDK 16299/LinearGradients/LinearGradients.xaml.cs

```diff
@@ -0,0 +1,263 @@
+        {
+            if (stop1.Color == _deepPink)
+            {
+                stop1.StartAnimation("Color", _changeStopToHoneydew);
+                stop2.StartAnimation("Color", _changeStopToDeepPink);
+            } else if (stop1.Color == _teal)
+            {
+                stop1.StartAnimation("Color", _changeStopToLightSkyBlue);
+                stop2.StartAnimation("Color", _changeStopToTeal);
+            }
+        }
+
+        // Switch the current colors of the two color stops in the linear gradient brush. Called when last XAML UI Element is entered by the mouse pointer
+        private void ReverseStopColorsOnEntered(CompositionColorGradientStop stop1, CompositionColorGradientStop stop2)
+        {
+            if (stop1.Color == _honeydew)
+            {            
+                stop1.StartAnimation("Color", _changeStopToDeepPink);
+                stop2.StartAnimation("Color", _changeStopToHoneydew);
+            } else if (stop1.Color == _lightSkyBlue)
+            {
+                stop1.StartAnimation("Color", _changeStopToTeal);
+                stop2.StartAnimation("Color", _changeStopToLightSkyBlue);
+            }
+        }
+
+        // Animate the offset of the visuals to move back and forth the length of the given visual that this method is called on
+        private void AnimateOffset(SpriteVisual target)
+        {
+            // The endpoint is the width of the inidividual sprite visual being targetted. The offset of the individual visuals will travel back and forth between these points
+            float endPoint = target.Size.X;
+            
+            // Create offset animation
+            ScalarKeyFrameAnimation offsetAnim = _compositor.CreateScalarKeyFrameAnimation();
+            offsetAnim.Duration = TimeSpan.FromSeconds(1);
+
+            // When the visual is on the left, we set the anchor point to the right edge of the visual and the offset animation to end on the right side
+            if (target.Offset.X == 0)
+            {
+                target.AnchorPoint = new Vector2(1f, 0f);
+                offsetAnim.InsertKeyFrame(1, endPoint);
+
+            }
+
+            // When the visual is on the right, we set the anchor point to the left edge of the visual and the offset animation to end on the left side
+            else if (target.Offset.X == endPoint)
+            {
+                target.AnchorPoint = new Vector2(0f, 0f);
+                offsetAnim.InsertKeyFrame(1, 0);
+            }
+            target.StartAnimation("Offset.X", offsetAnim);
+
+        }
+
+        // Method for setting animations for the first canvas. This method is unique in that it contains a call to switch the color stops when this canvas is entered by mouse pointer
+        private void Canvas1_PointerEntered(object sender, PointerRoutedEventArgs e)
+        {
+            AnimateScale(_vis1);
+            ChangeStopColorsOnEntered(_b1GradientStop1, _b1GradientStop2);
```

Comment:
Can you be consistent with the other _PointerEntered methods and do AnimateScale + AnimateOffset together rather than putting the ChangeStopColorsOnEntered in the middle. Consistency tells the reader "it's just like the others, except different in this one way".